udp/src/test/scala/docs/scaladsl/UdpSpec.scala

```diff
@@ -87,6 +88,49 @@ class UdpSpec
       sub.cancel()
     }
 
+    "send and receive messages with options" in {
+
+      // #bind-flow
+      val bindFlow: Flow[Datagram, Datagram, Future[InetSocketAddress]] =
+        Udp.bindFlow(bindToLocal, List(InetProtocolFamily()))
+      // #bind-flow
+
+      val ((pub, bound), sub) = TestSource
+        .probe[Datagram](system)
+        .viaMat(bindFlow)(Keep.both)
+        .toMat(TestSink.probe)(Keep.both)
+        .run()
+
+      val destination = bound.futureValue
+
+      {
+        // #send-datagrams
+        val destination = new InetSocketAddress("my.server", 27015)
+        // #send-datagrams
+        destination
+      }
+
+      // #send-datagrams
+      val messagesToSend = 100
+
+      // #send-datagrams
+
+      sub.ensureSubscription()
+      sub.request(messagesToSend)
+
+      // #send-datagrams
+      Source(1 to messagesToSend)
+        .map(i => ByteString(s"Message $i"))
+        .map(Datagram(_, destination))
+        .runWith(Udp.sendSink(List(InetProtocolFamily())))
+      // #send-datagrams
+
+      (1 to messagesToSend).foreach { _ =>
+        sub.requestNext()
+      }
+      sub.cancel()
```

Comment:
I'm struggling with this one: I can't see a way to hook into the underlying implementation to confirm options have been called on the socket. I checked the akka docs/tests hoping for inspiration but came up short.

Reply:
I did some reading on Java [UDP broadcast](https://support.biamp.com/Audia-Nexia/Control/TCP_and_UDP_discovery_methods#:~:text=UDP Broadcast is an automatic,and can be entered manually.) and modified your test to send UDP messages to a broadcast address instead of a specific hostname/ip. I [stole some code from here](https://www.baeldung.com/java-broadcast-multicast) to get the first broadcast address. If the broadcast socket option is not enabled then no messages are received, so I think this might be a valid test, can you confirm?

Reply:
I put up a PR on your PR branch: https://github.com/CptLemming/alpakka/pull/1

Reply:
Sorry for the delay; I did some testing with the code you put up and did confirm without the socket options the tests failed so it seemed a valid way to check this. Thanks!